web/server/codechecker_server/session_manager.py

```diff
@@ -480,6 +478,36 @@ def __update_groups(self, user_name, groups):
 
         return False
 
+    def __try_regex_groups(self, username):
+        """
+        Return a set of groups that the user belongs to, depending on whether
+        the username matches the regular expression of the group.
+
+        """
+        if not self.__is_method_enabled('regex_groups'):
+            return set()
+        regex_groups = self.__auth_config['method_regex_groups']\
+                           .get('groups', [])
+        if not regex_groups:
+            return set()
+
+        # Compile all regular expressions once and store them for later use.
+        try:
+            group_regexes_compiled = self.__group_regexes_compiled
+        except AttributeError:
+            d = dict()
+            for group_name, regex_list in regex_groups.items():
+                d[group_name] = [ re.compile(r) for r in regex_list ]
+            self.__group_regexes_compiled = d
+            group_regexes_compiled        = d
```

Comment:
For me this solution is a little bit hacky. It would be better to initalize the `__group_regexes_compiled` member field in the `__init__` constructor function instead of doing this:
https://github.com/Ericsson/codechecker/blob/75ef0ec24f9ef0f3ae6cd0e3fc57d04ab1114c21/web/server/codechecker_server/session_manager.py#L158

Reply:
Is the `SessionManager` object initialised once when the server starts? Or per connection? I am just trying to figure out if there is real benefit to pre-compiling the regular expressions.